starknet_programs/cairo2/factorial_tr.cairo

```diff
@@ -0,0 +1,26 @@
+#[starknet::interface]
+trait IFactorial<TContractState> {
+   fn factorial(self: @TContractState, n: felt252) -> felt252;
+}
+
+#[starknet::contract]
+mod Factorial {
+    #[storage]
+    struct Storage {
+    }
+
+    fn factorial_tr(acc: felt252, n: felt252) -> felt252 {
+        if n == 0 || n == 1 {
+            acc
+        } else {
+            factorial_tr(acc*n, n-1)
+        }
+    }
+
+    #[external(v0)]
+    impl Factorial of super::IFactorial<ContractState> {
+        fn factorial(self: @ContractState, n: felt252) -> felt252 {
+            factorial_tr(1, n)
+        }
+    }
+}
```

Comment:
Add newline at end of file

Reply:
thx

Reply:
I moved to a new computer and forgot to do that change

Reply:
done